src/storage-client/src/types/connections.rs

```diff
@@ -1016,13 +1016,52 @@ impl SshTunnel {
 
         // ...record it in the broker lookup, with a task that
         // will shutdown the session on drop.
-        let (tx, rx) = channel();
+        let (tx, mut rx) = channel::<()>();
         mz_ore::task::spawn(|| format!("{}_ssh_session", debug_str), async move {
-            let _: Result<(), _> = rx.await;
+            async fn wait_and_check(session: &openssh::Session) -> Result<(), openssh::Error> {
+                // Check the ssh tunnel status periodically.
+                tokio::time::sleep(std::time::Duration::from_secs(30)).await;
+                session.check().await
+            }
+            {
+                let check_fut = wait_and_check(&session);
```

Comment:
I believe you can replace this whole thing with this:

```rust
loop {
    match match tokio::time::timeout(Duration::from_secs(30), rx).await {
        Ok(_) => break,
        Err(_) => if let Err(e) = session.check().await {
             warn!("ssh tunnel at port {local_port} failed to check: {:#}", anyhow!(e));
        },
    }
};
```

If I'm reading it correctly it behaves identically

Reply:
yes it does!